### DISCUSSION TITLE
Adding or subtracting two sensors

### DISCUSSION THREAD

**Discussion by AdamSoddo** [dbuezas/lovelace-plotly-graph-card#55] [General]
I know that this can be done with virtual sensors inside Home Assistant, but I was hoping to do it only in my graph.

I would like to graph the difference (Datahttps://github.com/dbuezas/lovelace-plotly-graph-card/discussions?discussions_q=Adding%20or%20subtracting%20two%20sensors%201 - Data2 - Data3).  Is that possible with the lambda function? 

I am already using the code found here  [#35](https://github.com/dbuezas/lovelace-plotly-graph-card/discussions/35) to group data by minute to ensure that time mismatch between my data points are handled.   For this new subtraction I would need to keep this code to make sure we are subtracting data within the minute range on the x-axis.

Thanks

**Comment by dbuezas** [comment]
Hey Adam,
It is possible at the moment through a hack.

I'll consider more elegant ways if you can suggest how the yaml would look like.

## The hack
* Add the two entries, one for each entity
* in the first one, make a lambda that stores the data by writing `window.entity1 = {xs,ys}
* in the second one also add a lambda to pick  window.entity1 and do the math.
* The x coordinate of each data point won't match across different entities, so you'll have to account for that. Check the stacked bar graphs discussion for a solution to that.

Le me know if you don't succeed and I'll guide you when I'm back from vacations :)

**Reply by AdamSoddo** [reply]
Thanks for the quick response. I'll need a deeper tutorial about how to store data with the lambda function.  I have tried a number of approaches and I can't seem to reference the previously stored data.  Are you using a "BACKTICK" character  in your instructions? 

I am also struggling with math functions in the lambda. It seems like I'm working on the whole array of data (xs,ys)  That means I need to use array functions to add things together.  I am trying to use some examples here to fill in my gaps of knowledge about Java functions... but not doing great.  If anyone can point me to a stronger overview of the lambda function and what it can do I can read up.

Thanks in advance, and please don't worry about this while you are on vacation!

**Reply by dbuezas** [reply]
Sounds good!
If nobody does it earlier, I'll write the example when Im back.
I appreciate your consideration for my vacation :) I do have fun with this so no need to worry about me worrying 😁

**Comment by dbuezas** [comment]
⚠️ Attention: this is now possible and easy with filters.
